embassy-net/src/lib.rs

```diff
@@ -363,6 +366,56 @@ impl<D: Driver + 'static> Stack<D> {
         v4_up || v6_up
     }
 
+    /// Wait for the network stack to obtain a valid IP configuration.
+    /// Returns instantly if [`Stack::is_config_up`] returns `true`.
+    ///
+    /// ## Watch out:
```

Comment:
I think this comment can be removed, or perhaps simplified to something like "Ensure `Stack::run` has been called" and defer explaining why to  `run`'s documentation.